### PR TITLE
Add Code Graph OKRs to strategy page

### PR DESCRIPTION
**Pull request by jasonyavorsky**
To be merged once https://docs.google.com/document/d/1j_GuBOLXDVpzBzfCWJl-hz2rvqT--IPo3GYU07aG8HE/edit#bookmark=id.f3exy2b8hxym is agreed upon (please have discussions there, not here.)

### CODE REVIEW

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [inline content/company/strategy/code-graph/index.md]:
```suggestion
- Measure and increase [WAU](../../../bizops/user_definitions.md) for product areas that are important to paid adoption by 15%. For most teams this will require understanding and improving time-to-value as a driver for increasing active users. (All Teams)
```

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [inline content/company/strategy/code-graph/index.md]:
```suggestion
We are focused on several themes to help make our one year vision a reality. They are not listed in priority order. To see how these play into what any particular product area is delivering, take a look at the individual team strategy pages linked at the [top of this page](#code-graph).
```

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [inline content/company/strategy/code-graph/index.md]:
```suggestion
- Measure and increase [WAU](../../../bizops/user_definitions.md) for [customer facing product areas](https://github.com/sourcegraph/handbook/blob/main/data/product_areas.yml) by 15%. For most teams this will require understanding and improving time-to-value as a driver for increasing active users. (All Teams)
```

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [inline content/company/strategy/code-graph/index.md]: TODO replace?

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [inline content/company/strategy/code-graph/index.md:55]: @malomarrec @yinquanteo if you change the OKR in the FY22 OKRs doc please update it here also so this is ready to merge once approved.

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [inline content/company/strategy/code-graph/index.md]:
```suggestion
- Enable powerful automation and notification use cases e.g. by adding webhook and Slack integrations to Code Monitors and getting a Q1 upgrade commitment from 3 customers. ([Search Product](./search/index.md))
```

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [reply content/company/strategy/code-graph/index.md]:
```suggestion
- Make Sourcegraph Search work well with Salesforce-scale monorepos; some examples of achieving this could be P75 time to first result for unindexed search on monorepos larger than 200GB is < 10s and <2s for indexed, time to indexing is always <30 minutes, and we understand how search results should be ranked for monorepos. ([Search Core](./search/index.md))
```

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [inline content/company/strategy/code-graph/index.md]:
```suggestion
- Make Sourcegraph Search work well with Salesforce-scale monorepos; some examples of achieving this could be P75 time to first result for unindexed search on monorepos larger than 200GB is < 10s and <2s for indexed, time to indexing is always <30 minutes, and we understand how search results should be ranked for monorepos. ([Search Core](./search/index.md))
- Sourcegraph Cloud indexes public repositories globally from 4 non-GitHub code hosts. ([Search Core](./search/index.md))
```

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [inline content/company/strategy/code-graph/index.md]:
```suggestion
We also set our goals around these themes, sou can see current (Q4 2021) goals as a bullet list in each theme. The team listed after each goal is the primary team responsible for implementing the goal, but in the end we achieve these together and will support each other as needed to make them reality.
```

**Comment by jasonyavorsky** [comment]
@christinaforney @nicksnyder this is ready by for your :+1: to merge (no changes from what is agreed upon in the Google doc).

File: content/company/strategy/code-graph/index.md
Comment (nicksnyder) [inline content/company/strategy/code-graph/index.md]: This one is a bit confusing as far as how we are going to measure success.

> Q1 upgrade commitment from 3 customers.

What is the customer committing to upgrade here? Their instance or their usage of code monitors?

> e.g. by adding webhook and Slack integrations to Code Monitors

Is this just an example, or is adding this support a key part of the definition of success?

Given I don't understand the semantics of your intent here, here are some possible rewrites:
* Add webhook and Slack integrations to Code Monitors that solve the use case for customer(s) X, Y, Z
* 3 customers have committed to upgrading their Sourcegraph instance in Q1 so that they can use code monitors.

File: content/company/strategy/code-graph/index.md
Comment (nicksnyder) [inline content/company/strategy/code-graph/index.md]: > Improve precise code intelligence actions coverage by going from 3 to 5 officially ready languages (e.g., adding Kotlin and TS/JS)

This phrasing communicates that we have uncertainty about whether Kotlin and TS/JS are the right next two languages to pursue. Is this correct? If not, and if we have confidence, then maybe just make that more explicit in the KR (e.g. `Kotlin and TS/JS become officially supported for precise code intelligence`)

In any case, can you link to our list of officially ready languages (which hopefully also includes the definition of what officially ready means)?

File: content/company/strategy/code-graph/index.md
Comment (nicksnyder) [inline content/company/strategy/code-graph/index.md]: Do we not know what exactly scale problem(s) Salesforce is experiencing?

I would rather see the team commit to a single clear result here, and then update it later if it turns out we were wrong, instead of being non-committal about what success looks like.

Alternatively, you could turn this KR into something that could be validated with a Salesforce outcome. For example: `Salesforce technical win not blocked by any issues with search`

**Review by nicksnyder — APPROVED** [review]
Directionally, all these look good. My only comments are around the phrasing so it is clear what criteria need to be satisfied to declare success for a KR.

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [reply content/company/strategy/code-graph/index.md]: @benvenker can you confirm?

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [reply content/company/strategy/code-graph/index.md]: "Officially ready" wasn't a term of art in this case so removed it, it just means ones that work.

```suggestion
- Improve precise code intelligence actions coverage by going from 3 to 5 languages (adding Kotlin and TS/JS), representing the languages in ~50% of all actions. ([Code intelligence](./code-intelligence/index.md))
```

Comment (jasonyavorsky) [reply content/company/strategy/code-graph/index.md]: @macraig FYI on the update above, let me know if anything looks incorrect there and we can fix it.

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [reply content/company/strategy/code-graph/index.md]: To start I'll replace the customer name here with large-scale.

```suggestion
- Make Sourcegraph Search work well with large-scale monorepos; some examples of achieving this could be P75 time to first result for unindexed search on monorepos larger than 200GB is < 10s and <2s for indexed, time to indexing is always <30 minutes, and we understand how search results should be ranked for monorepos. ([Search Core](./search/index.md))
```

Comment (jasonyavorsky) [reply content/company/strategy/code-graph/index.md]: @lguychard could you take a look at the feedback above?

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [reply content/company/strategy/code-graph/index.md]: I believe it's just "upgrading to use" and we can say that more simply by "commitment to use".

```suggestion
- Enable powerful automation and notification use cases (e.g. by adding webhook and Slack integrations) to Code Monitors and getting a Q1 usage commitment from 3 customers. ([Search Product](./search/index.md))
```

@benvenker if this was wrong let me know/add a correction.

File: content/company/strategy/code-graph/index.md
Comment (jasonyavorsky) [inline content/company/strategy/code-graph/index.md]: Maybe this is a good update @lguychard ? It re-makes the criteria we discussed above part of the KR again, and removes the software one "we understand how.." which we can still do but not track as part of the key result.

```suggestion
- Make Sourcegraph Search work well with large-scale monorepos (time to first result for unindexed search on monorepos larger than 200GB is < 10s and <2s for indexed, and time to indexing is always <30 minutes). ([Search Core](./search/index.md))
```

Comment (jasonyavorsky) [reply content/company/strategy/code-graph/index.md]: Context feedback from @nicksnyder:

> Do we not know what exactly scale problem(s) they are experiencing?
>
> I would rather see the team commit to a single clear result here, and then update it later if it turns out we were wrong, instead of being non-committal about what success looks like.
> 
> Alternatively, you could turn this KR into something that could be validated with a customer outcome. For example: Key accounts technical wins not blocked by any issues with search

Comment (jasonyavorsky) [reply content/company/strategy/code-graph/index.md]: @lguychard looks like you are still out today, so I'm going to commit this suggestion and merge it. We can still update it when you get back.

Comment (lguychard) [reply content/company/strategy/code-graph/index.md]: This looks good to me. Thank you! (cc also @jjeffwarner)

Comment (jjeffwarner) [reply content/company/strategy/code-graph/index.md]: Works for me as well.

**Pull request merged**
